pre_commit/commands/run.py

```diff
@@ -37,7 +43,15 @@ def get_changed_files(new, old):
     )[1].splitlines()
 
 
-def get_filenames(args, include_expr, exclude_expr):
+def filter_filenames_by_types(filenames, types):
+    types = frozenset(types)
+    return tuple(
+        filename for filename in filenames
+        if tags_from_path(filename) >= types
```

Comment:
Yeah I do think it should be `AND` - I also think I should implement `exclude_types` as we have (`files`, `exclude`) but no way to do that with `types`.